seqr/views/utils/terra_api_utils_tests.py

```diff
@@ -0,0 +1,90 @@
+import mock
+import responses
+
+from django.test import TestCase
+from django.contrib.auth.models import User
+
+from seqr.views.utils.terra_api_utils import get_anvil_billing_projects, get_anvil_profile, has_anvil_session,\
+    list_anvil_workspaces, get_anvil_workspace_acl
+from settings import TERRA_API_ROOT_URL
+
+
+class TerraApiUtilsCase(TestCase):
+    fixtures = ['users']
+
+    def setUp(self):
+        google_url = 'https://oauth2.googleapis.com/token'
+        content = b'{"access_token":"ya29.c.KpMB4QdMNGgxWQvj3PxOgbQqfWQznoHuuoY03vN_A-W_W3vkdXJ6q-a539jSZEWc602cBL_RfNUH4GZV-d8rosfzQwBHoCVxfQs-Su2jBgKEdTui53jfjyU9T9LxshDl8ov31yk8Pi8f9D9ER7EUp-W1mP9gSy9e4YWkLb_q4hdJRR3BJywlQIWvPEfawpjC5Jpt_qGJ","expires_in":3599,"token_type":"Bearer"}'
+        responses.add(responses.POST, google_url, status = 200, body = content)
+        self.user = User.objects.get(email__iexact = 'test_user@test.com')
+        has_anvil_session(self.user)
+
+    @responses.activate
+    def test_get_billing_projects(self):
+        url = TERRA_API_ROOT_URL + 'api/profile/billing'
+        responses.add(responses.GET, url, status = 200, body = '[{"creationStatus": "Ready","projectName": "my-seqr-billing","role": "Owner"}]')
+
+        billing_projects = get_anvil_billing_projects(self.user)
+        self.assertEqual(len(billing_projects), 1)
+        self.assertEqual(billing_projects[0]['projectName'], 'my-seqr-billing')
+
+        responses.replace(responses.GET, url, status = 401)
+        with self.assertRaises(Exception) as ec:
+            _ = get_anvil_billing_projects(self.user)
+        self.assertEqual(str(ec.exception),
+            'Error: called Terra API "api/profile/billing" got status: {} with a reason: {}'.format(401, 'Unauthorized'))
+
+    @responses.activate
+    def test_get_anvil_profile(self):
+        url = TERRA_API_ROOT_URL + 'register'
+        responses.add(responses.GET, url, status = 200, body = '{"enabled":{"ldap":true,"allUsersGroup":true,"google":true},"userInfo": {"userEmail":"sf-seqr@my-seqr.iam.gserviceaccount.com","userSubjectId":"108344681601016521986"}}')
+
+        profile = get_anvil_profile(self.user)
+        self.assertDictEqual(profile, {'enabled': {'ldap': True, 'allUsersGroup': True, 'google': True},
+            'userInfo': {'userEmail': 'sf-seqr@my-seqr.iam.gserviceaccount.com', 'userSubjectId': '108344681601016521986'}})
+
+        responses.replace(responses.GET, url, status = 404)
+        with self.assertRaises(Exception) as ec:
+            _ = get_anvil_profile(self.user)
+        self.assertEqual(str(ec.exception),
+            'Error: called Terra API "register" got status: {} with a reason: {}'.format(404, 'Not Found'))
```

Comment:
please hardcode strings that can't change, do not use constants as formatting variables all it does is make the code less readable

Reply:
Okay